### DISCUSSION TITLE
v2.11.12 to v3.0.4 database migrations failed

### DISCUSSION THREAD

**Discussion by rhyser9** [netbox-community/netbox#7398] [Q&A] [answered]
Hi all,

I'm attempting to upgrade our instance of Netbox from v2.11.12 to v3.0.4. While running the `./upgrade.sh` script, it fails stating that the database migrations are out of sync. Upon investigating with `python3 netbox/manage.py showmigrations`, I receive the error `ImportError: cannot import name 'StaticSelect2' from 'utilities.forms' (/opt/netbox-3.0.4/netbox/utilities/forms/__init__.py)`.

Any ideas? Is this an issue with v3.0.4, or did am I missing something regarding our local installation? Details are as follows:

**Software Versions:**
```
OS: Ubuntu 20.04.1 LTS
Python: 3.8.10
pgSQL: 12.8
Redis: 5.0.7
```

**Upgrade Process:**
```
sudo wget https://github.com/netbox-community/netbox/archive/v3.0.4.tar.gz
sudo tar -xzf v3.0.4.tar.gz -C /opt
sudo ln -sfn /opt/netbox-3.0.4/ /opt/netbox

sudo cp /opt/netbox-2.11.12/local_requirements.txt /opt/netbox/
sudo cp /opt/netbox-2.11.12/netbox/netbox/configuration.py /opt/netbox/netbox/netbox/
sudo cp /opt/netbox-2.11.12/netbox/netbox/ldap_config.py /opt/netbox/netbox/netbox/

sudo cp -pr /opt/netbox-2.11.12/netbox/media/ /opt/netbox/netbox/
sudo cp -r /opt/netbox-2.11.12/netbox/scripts /opt/netbox/netbox/
sudo cp -r /opt/netbox-2.11.12/netbox/reports /opt/netbox/netbox/
sudo cp /opt/netbox-2.11.12/gunicorn.py /opt/netbox/

sudo /opt/netbox/upgrade.sh

sudo systemctl restart netbox netbox-rq

cp /opt/netbox/contrib/netbox-housekeeping.sh /etc/cron.daily/
```

**`upgrade.sh` Console Output:** https://gist.github.com/rhyser9/e1b1a365ff424bea80a000b6d1ec2228
**`showmigrations` Output:** https://gist.github.com/rhyser9/b391c7fc2e220b2e5bff14c3997b45ab

**Comment by rhyser9** [comment, marked as answer]
Upon further investigation, I think this may be due plugins; in particular, the `netbox-bgp` plugin (https://github.com/k01ek/netbox-bgp). I will attempt to disable/remove all plugins (they aren't in use, were just for testing) then try the upgrade process again.

**Reply by rhyser9** [reply]
Uninstalled netbox-bgp plugin and removed it from `configuration.py`. Re-ran the upgrade script and it worked without issue.

**Reply by jeremystretch** [reply]
Thanks for following up with your solution!

**Reply by candlerb** [reply]
The error saying "database migrations are out of sync" is misleading, and I've already raised that as #7319.  If you'd seen the raw `ImportError` exception up-front, it would probably have been clearer what was happening.
